All pens embed

Mehnix opened the pull request:
Pen bombs embed into people
Regular pens don't

## About the PR
Now regular pens embed like their bomb counterparts. Basic Thrown pens now do 1 damage.

## Why / Balance
Notably harder to tell apart pens and pen bombs.

## Technical details
Moves thrown damage to the base "Pen" entity.
Removes PenEmbeddable Abstract entity, because it is somewhat redundant if regular pens can already embed.
Fancy Pens still do 3 damage.

## Media
<img width="211" height="234" alt="image" src="https://github.com/user-attachments/assets/f7ff14fa-f18f-43b6-aee3-2e4c93f1789b" />

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
The PenEmbeddable entity prototype has been removed, since all pens now embed. Any entities that used this as a parent should instead use Pen or BasePenFancy

**Changelog**
:cl:
- tweak: All pens now embed when thrown.

Mehnix committed:
Embed for all

slarticodefast commented:
Duplicate of https://github.com/space-wizards/space-station-14/pull/32831 which was declined in the past.

Mehnix commented:
> Duplicate of #32831 which was declined in the past.

This PR came out of a discussion of the loss of the meta shield offering rules legal ways to check syndicate stealth items in unfun ways, such as throwing a pen to see if it embeds, indicating it is a pen bomb.

As removing a pen bomb's ability to embed would ruin the item, something along this line seems more reasonable in making stealth items act like their real counterparts. Exact specifics may vary, could always fall out after some time. Personally I don't see the harm in it, I set it to 0.1 damage to reduce any real griefing potential.

Cojoke-dot commented:
I don't think this would do shit for a hiding a penbomb of someone actually cares to check. If they check your pen they click it to see if its a dagger then throw it incase its a bomb. The only thing throwing pens like this is good for is fun

Mehnix commented:
> I don't think this would do shit for a hiding a penbomb of someone actually cares to check. If they check your pen they click it to see if its a dagger then throw it incase its a bomb. The only thing throwing pens like this is good for is fun

A regular security officer arming a pen and then throwing it, knowing that it might be a lethal explosive is a far bigger concern in my opinion than someone possessing it. That's like pulling the pin on a grenade to see if its real. At least on MRP that seems like it'd be against basic code of conduct for a secoff.

Mehnix commented:
> If someone is already checking pens the first thing they are going to do is click it because edaggers are a thing

This would likely require a seperate and far more involved PR that could, for example, obfuscate the ability to use stealth items to those who haven't mechanically "learned" how they work (the person who bought it, and anyone they deliberately demonstrate it to for example would know). 

That doesn't invalidate this PR, as stealth items and regular items should act the same. Making it so at least a secoff has to arm an actual live grenade to tell the difference is better than nothing.

beck-thompson commented on Resources/Prototypes/Entities/Objects/Misc/pen.yml:
```diff
@@ -57,6 +51,16 @@
         restitution: 0.3
         friction: 0.2
 
+- type: entity
```
I don't think this is good to have, players could still metagame how much damage it does (It wouldn't surprise me at this point!)

3 vs 0.1 is also very minor, I say just make 1 dmg for all pens

Mehnix committed:
Unify all thrown pen damage

Removes PenEmbeddable Abstract Entity as it is redundant. Those that parented to it now just parent to Pen.

Mehnix committed:
Fancy Pens deserve that damage bonus

Keeps an abstract entity for fancier pens to parent off of. Someone might do different fonts in the future, and them doing more damage is funny.

Princess-Cheeseballs reviewed (approved):
You ever have the approve window open after reviewing something and forget to hit enter for like an entire day?

Anyways thank you for your contribution!
